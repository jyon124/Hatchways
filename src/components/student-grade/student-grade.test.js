import React from "react";
import { mount } from "enzyme";
import { waitFor } from "@testing-library/react";
import StudentGrade from "./student-grade";

describe("Student Grade Component Tests", () => {
    it("Renders a Student Grade component", async () => {
        expect.assertions(3);
        const mockIdx = 1;
        const mockGrade = "94";
        const wrapper = mount(<StudentGrade grade={mockGrade} idx={mockIdx} />);
        await waitFor(() => {
            expect(wrapper.props().grade).toBe(mockGrade);
            expect(wrapper.props().idx).toBe(mockIdx);
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
