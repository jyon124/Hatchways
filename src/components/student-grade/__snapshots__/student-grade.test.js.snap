// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Grade Component Tests Renders a Student Grade component 1`] = `"<div class=\\"grade\\">Test2: 94%</div>"`;
